### PR TITLE
Add azure openai example

### PR DESCRIPTION
**Pull request by serena-ruan**
## What changes are proposed in this pull request?
Add azure openai example

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [x] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: examples/openai/azure_openai.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import mlflow
+import openai
+import os
+import pandas as pd
+
+os.environ["OPENAI_API_KEY"] = "<YOUR AZURE OPENAI KEY>"
```
Comment (BenWilson2) [inline examples/openai/azure_openai.py:6]: Can we remove the key declaration in code and instead add a block comment about setting these keys through a terminal?

File: examples/openai/azure_openai.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+os.environ["OPENAI_API_BASE"] = "<YOUR AZURE OPENAI BASE>"
```
Comment (BenWilson2) [inline examples/openai/azure_openai.py:7]: Can we add a comment with an example of the formatting specifics for the url?

File: examples/openai/azure_openai.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+os.environ["OPENAI_API_VERSION"] = "2023-05-15"
+os.environ["OPENAI_API_TYPE"] = "azure"
+
+with mlflow.start_run():
+    model_info = mlflow.openai.log_model(
+        # For Azure OpenAI, model doesn't matter in practice
```
Comment (BenWilson2) [inline examples/openai/azure_openai.py:13]: Isn't this mapped to `engine` and will select among the different offerings?

File: examples/openai/azure_openai.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        model="gpt-3.5-turbo",
```
Comment (BenWilson2) [inline examples/openai/azure_openai.py:14]:
```suggestion
        model="gpt-35-turbo",
```
periods are removed in the engine version designators in azure open ai

File: examples/openai/azure_openai.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        task=openai.ChatCompletion,
+        artifact_path="model",
+        messages=[{"role": "user", "content": "Tell me a joke about {animal}."}],
+        deployment_id="<YOUR AZURE DEPLOYMENT ID (ALSO CALLED DEPLOYMENT NAME)>",
+    )
+
```
Comment (BenWilson2) [inline examples/openai/azure_openai.py:20]: Can we show a native model load example? 
`native_model = mlflow.openai.load_model()`
`completion = openai.ChatCompletion.create(...)`

**Review by BenWilson2 — APPROVED** [review]
Some minor comments, otherwise LGTM! TY @serena-ruan

File: examples/openai/azure_openai.py (hunk 1 above)
Comment (harupy) [reply examples/openai/azure_openai.py:6]: +1. This can lead to leaking credentials.

File: examples/openai/azure_openai.py (hunk 3 above)
Comment (serena-ruan) [reply examples/openai/azure_openai.py:13]: I'm not sure about this, I can remove the comment. But looks like in order to use Azure OpenAI you need to deploy a model first, and that deployment_id/name is used against request.

**Review by BenWilson2 — APPROVED** [review]
Stamped again! TY!

**Pull request merged**